api/envoy/api/v2/route/route_components.proto

```diff
@@ -1061,7 +1045,7 @@ message RetryPolicy {
   // `upstream.base_retry_backoff_ms` runtime parameter. The default maximum interval is 10 times
   // the base interval. The documentation for :ref:`config_http_filters_router_x-envoy-max-retries`
   // describes Envoy's back-off algorithm.
-  RetryBackOff retry_back_off = 8;
+  core.BackoffStrategy retry_back_off = 8;
```

Comment:
Even though this is wire compatible I think this is a breaking change since it's a compile breakage. @htuch? Should we instead have some type of transform message that changes the type at the next major version?

Reply:
Yeah, we don't allow compile time breakages anymore, so we can't do this kind of cleanup within a major version. You will need to do the new field / old field deprecation dance.

Reply:
@htuch for this case I think we could just have a "type change" annotation which is done during major version upgrade? That would be less painful. Should we track that?

Reply:
Type changes don't work today with the automatic upgrade/downgrade. It would be pretty challenging to account for them, as we do, to a larger extent, rely on wire duck typing.